### PR TITLE
Support incontiguous logits and bitmask.

### PR DESCRIPTION
**Pull request by hnyls2002**
This PR supports incontiguous logits and bitmasks when applying token bitmasks (The vocab stride still requires to be 1)

### CODE REVIEW

**Review by Ubospica — APPROVED** [review]
LGTM! We can add a test and merge it

**Comment by Ubospica** [comment]
We can merge it for now and add tests later
